src/dk.nita.saml20/dk.nita.saml20/Utils/BouncyCastleHelper.cs

```diff
@@ -85,6 +85,34 @@ public static byte[] DecryptAssertionWithAesGcm(EncryptedData encryptedData, byt
             cipher.DoFinal(output, len);
 
             return output;
+        }*/
+        
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="encryptedData"></param>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public static byte[] Decrypt(EncryptedData encryptedData, byte[] key)
+        {
+            // Base64 decode encrypted data
+            var nonceCipherValue = encryptedData.CipherData.CipherValue;
+
+            // separate nonce and ciphertextTag
+            const int nonceSize = 12;
+            var nonce = new byte[nonceSize];
+            Array.Copy(nonceCipherValue, 0, nonce, 0, nonceSize);
+            var ciphertextTag = new byte[nonceCipherValue.Length - nonceSize];
+            Array.Copy(nonceCipherValue, nonceSize, ciphertextTag, 0, ciphertextTag.Length);
+
+            var gcmBlockCipher = new GcmBlockCipher(new AesEngine());
+            gcmBlockCipher.Init(false, new AeadParameters(new KeyParameter(key), 128, nonce));
```

Comment:
```suggestion
            gcmBlockCipher.Init(false, new AeadParameters(new KeyParameter(key), macSize * 8, nonce));
```
Maybe extract mac size in bytes constant, const int macSize = 16;